google/generativeai/caching.py

```diff
@@ -0,0 +1,260 @@
+# -*- coding: utf-8 -*-
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import dataclasses
+import datetime
+from typing import Any, Iterable, Optional
+
+from google.generativeai import protos
+from google.generativeai.types.model_types import idecode_time
+from google.generativeai.types import caching_types
+from google.generativeai.types import content_types
+from google.generativeai.utils import flatten_update_paths
+from google.generativeai.client import get_default_cache_client
+
+from google.protobuf import field_mask_pb2
+import google.ai.generativelanguage as glm
+
+
+@dataclasses.dataclass
+class CachedContent:
```

Comment:
I can't read the "everyting on one line" printout.

please use: https://github.com/google-gemini/generative-ai-python/blob/7b9758f54180b525d355393302100b0dea082a12/google/generativeai/string_utils.py#L52

Reply:
Done!